src/main/java/org/elasticsearch/indices/IndicesService.java

```diff
@@ -467,7 +464,7 @@ public void deleteClosedIndex(String reason, IndexMetaData metaData) {
      * Deletes the index store trying to acquire all shards locks for this index.
      * This method will delete the metadata for the index even if the actual shards can't be locked.
      */
-    public void deleteIndexStore(String reason, IndexMetaData metaData) throws IOException {
+    public void deleteIndexStore(String reason, IndexMetaData metaData, ClusterState clusterState) throws IOException {
```

Comment:
can we please update the signature here to reflect what state this is? I am not sure if we should do this here. The internal state wasn't good enough?